src/tokens/ERC6909.sol

```diff
@@ -0,0 +1,108 @@
+
+    /*//////////////////////////////////////////////////////////////
+                             ERC6909 STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    mapping(uint256 => uint256) public totalSupply;
+
+    mapping(address => mapping(uint256 => uint256)) public balanceOf;
+
+    mapping(address => mapping(address => mapping(uint256 => uint256))) public allowance;
+
+    mapping(address => mapping(address => bool)) public isOperator;
+
+    /*//////////////////////////////////////////////////////////////
+                              ERC6909 LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function transfer(
+        address receiver,
+        uint256 id,
+        uint256 amount
+    ) public virtual returns (bool) {
+        balanceOf[msg.sender][id] -= amount;
+        balanceOf[receiver][id] += amount;
+        emit Transfer(msg.sender, msg.sender, receiver, id, amount);
+        return true;
+    }
+
+    function transferFrom(
+        address sender,
+        address receiver,
+        uint256 id,
+        uint256 amount
+    ) public virtual returns (bool) {
+        if (msg.sender != sender && !isOperator[sender][msg.sender]) {
+            uint256 allowed = allowance[sender][msg.sender][id];
+            if (allowed != type(uint256).max) allowance[sender][msg.sender][id] = allowed - amount;
+        }
+        balanceOf[sender][id] -= amount;
+        balanceOf[receiver][id] += amount;
+        emit Transfer(msg.sender, sender, receiver, id, amount);
+        return true;
+    }
+
+    function approve(
+        address spender,
+        uint256 id,
+        uint256 amount
+    ) public virtual returns (bool) {
+        allowance[msg.sender][spender][id] = amount;
+        emit Approval(msg.sender, spender, id, amount);
+        return true;
+    }
+
+    function setOperator(address operator, bool approved) public virtual {
+        isOperator[msg.sender][operator] = approved;
+        emit OperatorSet(msg.sender, operator, approved);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              ERC165 LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) {
+        return
+            interfaceId == 0x01ffc9a7 || // ERC165 Interface ID for ERC165
+            interfaceId == 0xb2e69f8a; // ERC165 Interface ID for ERC6909
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        INTERNAL MINT/BURN LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function _mint(
+        address receiver,
+        uint256 id,
+        uint256 amount
+    ) internal virtual {
+        balanceOf[receiver][id] += amount;
+        totalSupply[id] += amount;
```

Comment:
can this not be unchecked?

Reply:
(same spacing change requested here too)

Reply:
Fixed. The `receiver` balance increase on `transfer` and `transferFrom` can also be unchecked. Thoughts on unchecking those as well?

Reply:
yup lets uncheck everything you think is safe, just add a note explaining why like in ERC20 (https://github.com/transmissions11/solmate/blob/fadb2e2778adbf01c80275bfb99e5c14969d964b/src/tokens/ERC20.sol#L186-L187)